gremlin-go/driver/gorillaTransporter.go

```diff
@@ -68,16 +73,47 @@ func (transporter *gorillaTransporter) Write(data []byte) (err error) {
 	return err
 }
 
-func (transporter *gorillaTransporter) Read() (bytes []byte, err error) {
+func (transporter *gorillaTransporter) Read() ([]byte, error) {
 	if transporter.connection == nil {
-		err = transporter.Connect()
+		err := transporter.Connect()
 		if err != nil {
-			return
+			return nil, err
+		}
+	}
+	transporter.connection.SetPongHandler(func(string) error {
+		err := transporter.connection.SetReadDeadline(time.Now().Add(time.Second))
```

Comment:
`time.Now().Add(time.Second)`
Is this a from a convention or rule?

Reply:
It just seemed like a good amount of time